pkg/controller/promotionreconciler/reconciler.go

```diff
@@ -239,3 +260,59 @@ func configIndexFn(in cioperatorapi.ReleaseBuildConfiguration) []string {
 func configIndexKeyForIST(ist *imagev1.ImageStreamTag) string {
 	return ist.Namespace + "/" + ist.Name
 }
+
+func handleCIOpConfigChange(registryClient ctrlruntimeclient.Client,
+	ciOperatorConfigGetter ciOperatorConfigGetter,
+	prowJobEnqueuer prowjobreconciler.Enqueuer,
+	githubClient githubClient,
+	delta agents.IndexDelta,
+	log *logrus.Entry) error {
+	log = log.WithField("indexKey", delta.IndexKey)
+	log.Debug("Handling CIOpConfig change")
+	// We only care about new additions
+	if len(delta.Added) == 0 {
+		return nil
+	}
+	slashSplit := strings.Split(delta.IndexKey, "/")
+	if len(slashSplit) != 2 {
+		return fmt.Errorf("got an index delta event with a key that is not a valid namespace/name identifier: %s", delta.IndexKey)
+	}
+	namespace, name := slashSplit[0], slashSplit[1]
```

Comment:
Isn't the key the `organization`/`repository` of the configuration?

Reply:
This one confused me too. I might be still wrong now.

The key is determined by the `indexFunc` when adding the index.

https://github.com/openshift/ci-tools/pull/3270/files#diff-584fd395d199b51f651a3b4528bc23bfd07f3190509d659a1c1ad53d5f5bd248R58

which is `istRef.ISTagName()`
https://github.com/openshift/ci-tools/pull/3270/files#diff-584fd395d199b51f651a3b4528bc23bfd07f3190509d659a1c1ad53d5f5bd248R255